core/shared/src/main/scala/scalaz/zio/ZIO.scala

```diff
@@ -1140,7 +1153,7 @@ trait ZIOFunctions extends Serializable {
   final def superviseWith[R >: LowerR, E <: UpperE, A](
     zio: ZIO[R, E, A]
   )(supervisor: Iterable[Fiber[_, _]] => UIO[_]): ZIO[R, E, A] =
-    new ZIO.Supervise(zio, supervisor)
+    zio.ensuring(fibers.flatMap(supervisor(_))).supervised
```

Comment:
This is so much cleaner than before. The old `ZIO.Supervise` was conflating 2 things.